typing/subst.mli

```diff
@@ -29,22 +28,36 @@ type t
    Indeed, non-variable node of a type are duplicated, with their
    levels set to generic level.  That way, the resulting type is
    well-formed (decreasing levels), even if the original one was not.
-*)
 
-val identity: t
+   In the presence of local substitutions for module types, a substitution for a
+   type expression may fail to produce a well-formed type. In order to confine
+   this issue to local substitution, the type of substitutions is split into a
+   standard and local variant. Only local substitution may expand a module type
+   path into a generic module type. *)
+
+(** Type familly for substitutions *)
+type 'a s
+
+type t = [`safe] s
+(** Standard substitution*)
+
+type local = [`local] s
+(** Local substitution *)
```

Comment:
Naming proposal: some substitutions are "safe" (they always preserve well-formedness of the objects they operate on), some are "unsafe" (they may introduce ill-formed terms, which are detected at runtime). (Gets rid of "local" here, which is good).

Reply:
Unsafe was the first name of those variants, I have reverted the interface to this naming scheme.

Reply:
On reading the file below, I find `unsafe` a bit unclear, I would prefer `unsafe_subst` or just ``[`Unsafe] subst``. Maybe just ``type safe = [`Safe]``, and then `safe subst`?

Reply:
Nitpick: now you define `type unsafe` but there is no corresponding `type safe` alias, which could be slightly annoying for someone who would want to use this type explicitly. Could you introduce a `safe` type?